corehq/apps/linked_domain/tests/test_dbaccessors.py

```diff
@@ -12,52 +24,154 @@ class TestGetAvailableDomainsToLink(TestCase):
     def setUpClass(cls):
         super(TestGetAvailableDomainsToLink, cls).setUpClass()
 
-        cls.test_domain_obj1 = create_domain('test1')
-        cls.test_domain_obj2 = create_domain('test2')
-        cls.upstream_domain_obj = create_domain('upstream')
-        cls.downstream_domain_obj = create_domain('downstream')
+        # Setup non-enterprise subscription
+        cls.non_enterprise_domain_1 = generator.arbitrary_domain()
+        cls.non_enterprise_domain_2 = generator.arbitrary_domain()
+        cls.non_enterprise_account, _ = BillingAccount.get_or_create_account_by_domain(
+            cls.non_enterprise_domain_1.name,
+            created_by='user@test.com'
+        )
+        cls.non_enterprise_account.save()
+        cls._add_domain_to_account(cls.non_enterprise_domain_1.name,
+                                   cls.non_enterprise_account,
+                                   SoftwarePlanEdition.COMMUNITY)
+        cls._add_domain_to_account(cls.non_enterprise_domain_2.name,
+                                   cls.non_enterprise_account,
+                                   SoftwarePlanEdition.COMMUNITY)
+
+        # Setup enterprise subscription
+        cls.enterprise_domain_1 = generator.arbitrary_domain()
+        cls.enterprise_domain_2 = generator.arbitrary_domain()
+        cls.enterprise_account, _ = BillingAccount.get_or_create_account_by_domain(
+            cls.enterprise_domain_1.name,
+            created_by='user@test.com'
+        )
+        cls.enterprise_account.save()
+        cls._add_domain_to_account(cls.enterprise_domain_1.name,
+                                   cls.enterprise_account,
+                                   SoftwarePlanEdition.ENTERPRISE)
+
+        cls._add_domain_to_account(cls.enterprise_domain_2.name,
+                                   cls.enterprise_account,
+                                   SoftwarePlanEdition.ENTERPRISE)
 
         cls.user = WebUser.create(
-            domain=cls.test_domain_obj1.name,
```

Comment:
Naming: I'd keep the "_obj" since it's already being used, see [here](https://github.com/dimagi/commcare-hq/blob/42173e1511bbb9d86536273687219455c579dc06/corehq/apps/domain/models.py#L293-L300).

Reply:
Good point. I made that change and used `create_domain` instead of `arbitrary_domain` after our book club conversation this morning.